### PR TITLE
New tests for NotebookServiceImpl module

### PR DESCRIPTION
**Pull request by ranasaria**
This PR adds new tests for the NotebookServiceImpl module. It also has some dead code removal and a minor bug fix for this module. I am still going to add a test or two to cover a few error paths, but It would be good to get some feedback pouring in even as I add those.

Tests: Ran all unit tests.

Coverage: 
[Code coverage report for workbench_services_notebook_browser_notebookServiceImpl.ts.pdf](https://github.com/microsoft/azuredatastudio/files/4892076/Code.coverage.report.for.workbench_services_notebook_browser_notebookServiceImpl.ts.pdf)

### CODE REVIEW

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts [hunk 1]
```diff
@@ -317,12 +314,10 @@ export class NotebookService extends Disposable implements INotebookService {
 	get onNotebookEditorAdd(): Event<INotebookEditor> {
 		return this._onNotebookEditorAdd.event;
 	}
+
 	get onNotebookEditorRemove(): Event<INotebookEditor> {
 		return this._onNotebookEditorRemove.event;
 	}
-	get onCellChanged(): Event<INotebookEditor> {
```
Comment (ranasaria) [inline src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:323]: This was deadcode with no reference to listeners being registered or the emitter behind this event being fired anywhere in the code.

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts [hunk 2]
```diff
@@ -295,14 +292,14 @@ export class NotebookService extends Disposable implements INotebookService {
...
 	async getOrCreateNotebookManager(providerId: string, uri: URI): Promise<INotebookManager> {
 		if (!uri) {
-			throw new Error(localize('notebookUriNotDefined', "No URI was passed when creating a notebook manager"));
+			throw new Error(NotebookUriNotDefined);
 		}
 		let uriString = uri.toString();
 		let managers: INotebookManager[] = this._managersMap.get(uriString);
 		// If manager already exists for a given notebook, return it
 		if (managers) {
-			let index = firstIndex(managers, m => m.providerId === providerId);
```
Comment (ranasaria) [inline src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:304]: firstIndex api in vscode arrays module is deprecated and so replacing with their suggesion of using Array.findIndex

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts [hunk 3]
```diff
@@ -33,7 +33,6 @@ import { Schemas } from 'vs/base/common/network';
 import { ILogService } from 'vs/platform/log/common/log';
 import { toErrorMessage } from 'vs/base/common/errorMessage';
 import { NotebookChangeType } from 'sql/workbench/services/notebook/common/contracts';
-import { find, firstIndex } from 'vs/base/common/arrays';
```
Comment (ranasaria) [inline src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:36]: Both these apis are deprecated by vscode guys replacing its usage with their suggesion of Array.find and Array.findIndex for these.

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts [hunk 4]
```diff
@@ -86,27 +85,28 @@ class ProviderDescriptor {
...
 }
 
+export const NotebookUriNotDefined = localize('notebookUriNotDefined', "No URI was passed when creating a notebook manager");
+
 export class NotebookService extends Disposable implements INotebookService {
 	_serviceBrand: undefined;
 
-	private _providersMemento: Memento;
-	private _trustedNotebooksMemento: Memento;
+	private _providersMemento: Memento = new Memento('notebookProviders', this._storageService);
```
Comment (ranasaria) [inline src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:93]: moved intialization of these fields from within constructor to here as was done with most other properties in this class for consistency.

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts [hunk 5]
```diff
@@ -341,31 +336,31 @@ export class NotebookService extends Disposable implements INotebookService {
...
 	}
 
 	renameNotebookEditor(oldUri: URI, newUri: URI, currentEditor: INotebookEditor): void {
 		let oldUriKey = oldUri.toString();
 		if (this._editors.has(oldUriKey)) {
 			this._editors.delete(oldUriKey);
-			currentEditor.notebookParams.notebookUri = newUri;
-			this._editors.set(newUri.toString(), currentEditor);
+			currentEditor.notebookParams.notebookUri = newUri; //currentEditor.id gets this value as a string
+			this._editors.set(currentEditor.id, currentEditor);
```
Comment (ranasaria) [inline src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:353]: INotebookEditor's get id() is implemented as notebookParams.notebookUri.toString(), so this is a no-op change, but since all add, remove find uses editor.id as key for the Map, keeping it consistent here makes the code more readable.

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts [hunk 6]
```diff
@@ -341,31 +336,31 @@ export class NotebookService extends Disposable implements INotebookService {
...
 			this._onNotebookEditorRename.fire(currentEditor);
 		}
 	}
 
+	listNotebookEditors(): INotebookEditor[] {
+		let editors = [];
```
Comment (ranasaria) [inline src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:359]: Moved the method as wanted to keep the crud methods for noteboodEditor together.

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts [hunk 7]
```diff
@@ -375,11 +370,11 @@ export class NotebookService extends Disposable implements INotebookService {
 	private sendNotebookCloseToProvider(editor: INotebookEditor): void {
 		let notebookUri = editor.notebookParams.notebookUri;
 		let uriString = notebookUri.toString();
-		let manager = this._managersMap.get(uriString);
-		if (manager) {
+		let managers = this._managersMap.get(uriString);
```
Comment (ranasaria) [inline src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:373]: manager was a minnomer becasue at every entry in _managersMap we store an array or managers as value and not a single manager. So fixing the name makes the code read better.

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts [hunk 8]
```diff
@@ -389,10 +384,10 @@ export class NotebookService extends Disposable implements INotebookService {
 	private async doWithProvider<T>(providerId: string, op: (provider: INotebookProvider) => Thenable<T>): Promise<T> {
 		// Make sure the provider exists before attempting to retrieve accounts
 		let provider: INotebookProvider = await this.getProviderInstance(providerId);
-		return op(provider);
+		return await op(provider);
```
Comment (ranasaria) [inline src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:387]: since 'op' is thenable it should be await to avoid any weird timing issue. So this is a bug fix that may not have surfaced into an issue.

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts [hunk 9]
```diff
@@ -405,7 +400,7 @@ export class NotebookService extends Disposable implements INotebookService {
 				} catch (error) {
 					this._logService.error(error);
 				}
-				instance = await this.waitOnProviderAvailability(providerDescriptor);
+				instance = await this.waitOnProviderAvailability(providerDescriptor, timeout);
```
Comment (ranasaria) [inline src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:403]: timeout parameter to getProviderInstance method was being left unused early which was a bug.

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts [hunk 10]
```diff
@@ -541,7 +537,7 @@ export class NotebookService extends Disposable implements INotebookService {
 					await this.updateTrustedCache();
 				} else {
 					// Only save as trusted if the associated notebook model is trusted
-					let notebook = find(this.listNotebookEditors(), n => n.id === notebookUriString);
+					let notebook = this.listNotebookEditors().find(n => n.id === notebookUriString);
```
Comment (ranasaria) [inline src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:540]: remove deprecated use of find method from vscode array

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts [hunk 11]
```diff
@@ -532,7 +528,7 @@ export class NotebookService extends Disposable implements INotebookService {
 			// 3. Not already saving (e.g. isn't in the queue to be cached)
 			// 4. Notebook is trusted. Don't need to save state of untrusted notebooks
 			let notebookUriString = notebookUri.toString();
-			if (changeType === NotebookChangeType.Saved && firstIndex(this._trustedCacheQueue, uri => uri.toString() === notebookUriString) < 0) {
+			if (changeType === NotebookChangeType.Saved && this._trustedCacheQueue.findIndex(uri => uri.toString() === notebookUriString) < 0) {
```
Comment (ranasaria) [inline src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:531]: remove deprecated use of firstIndex method from vscode arrays

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts (hunk 8 above)
Comment (Charles-Gagnon) [reply src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:387]: This is doing nothing except adding extra overhead. If a user wasn't .then/await'ing the result then this await is just causing it to need to re-wrap the result in a Promise.

Now if callers weren't properly awaiting the call to doWithProvider then that's a separate issue, but this change should be reverted.

File: src/sql/workbench/contrib/notebook/test/testCommon.ts [hunk 12]
```diff
@@ -14,24 +16,39 @@ import { TestThemeService } from 'vs/platform/theme/test/common/testThemeService
...
 import { TestStorageService } from 'vs/workbench/test/common/workbenchTestServices';
 
+/**
+ *	Helper method to build get a spied object one can use to listen on using setups, returns and callback and the returned mock object
+ *
+ * @param object - the object to be spied upon
+ * @param behavior - the TypeMoq.MockBehavior value. Default is Loose
+ * @param callBase - whether to forward all calls by default to the base object. Default is true.
+ */
+export function getMockAndSpy<U>(object: U, behavior: TypeMoq.MockBehavior = TypeMoq.MockBehavior.Loose, callBase: boolean = true) {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/notebook/test/testCommon.ts:26]: return type for this

But I also don't see the value in this function. This isn't doing anything helpful in my eyes - there's no logic here since the user already passes in the required parameters and creating a mock is a 1 or 2 line thing. Nothing we should care about trying to consolidate IMO. 

And actually - do you even use this anywhere? I don't see it being used in this PR at all...

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts [hunk 13]
```diff
@@ -437,26 +430,26 @@ export class NotebookService extends Disposable implements INotebookService {
...
-			return new RenderMimeRegistry({
+			this._mimeRegistry = new RenderMimeRegistry({
 				initialFactories: standardRendererFactories
 			});
 		}
 		return this._mimeRegistry;
 	}
 
-	private get providersMemento(): NotebookProvidersMemento {
+	protected get providersMemento(): NotebookProvidersMemento {
```
Comment (anthonydresser) [inline src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:440]: Why are we just making a bunch of things protected? Private vs Protected is a design decision. Testing shouldn't affect that. If you need to change this visiblity to test it, your testing is probably wrong. There's a reason why in c# its considered bad practice to just make everything internal.

Comment (ranasaria) [reply src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:440]: It was charles' suggestion to make private fields to protected for testing purposes. C# has friend that circumvents this issue.

Comment (anthonydresser) [reply src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:440]: I very much disagree with that, like I said, private vs protected is a design decision not a testing decision. If someone else came and extended this class how are they supposed to know what methods were meant to be used by extended classes vs ones that were just made that way for testing?

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts [hunk 14]
```diff
@@ -295,14 +292,14 @@ export class NotebookService extends Disposable implements INotebookService {
...
+			throw new Error(NotebookUriNotDefined);
 		}
 		let uriString = uri.toString();
 		let managers: INotebookManager[] = this._managersMap.get(uriString);
 		// If manager already exists for a given notebook, return it
 		if (managers) {
-			let index = firstIndex(managers, m => m.providerId === providerId);
-			if (index && index >= 0) {
+			let index = managers.findIndex(m => m.providerId === providerId);
+			if (index >= 0) {
```
Comment (ranasaria) [inline src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:302]: (index && index >= 0) check was a pre-existing bug that I uncovered in testing. With code the way it was even if we had previously created and cached a manager in this._managersMap and managers.findIndex returned 0, we would not use that manager at 0th location but instead create a new one which is not what the code is intending to do.

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts (hunk 13 above)
Comment (Charles-Gagnon) [reply src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:440]: So let me rephrase what Arvind said with what I was actually saying : 

1. What Anthony said is 100% correct and I agree with it all
1. But I'm also aware that we have not been great about designing components in such a way that they can be easily testable without either a significant refactor, unless we modify private internals. For example some of our stuff is tightly coupled to the UI and so trying to make a test for that ends up being very difficult outside of a smoke test.

Now sure, we could (and have) just access these members via bracket notation. I'm hesitant to do this for two reasons
1. You don't get the compile time errors when a field is changed (so if it's renamed the tests would happily still try to access the old name and only fail then)
1. People modifying the class have no idea that other stuff is using these, which ends up being a pain when you make changes to what should be an implementation detail only to find out you need to go and update a bunch of tests that were modifying that state directly

There's a lot of nuance here - but I think at the end my thoughts are : 

1. We should always be striving to test things without needing to modify/access their internal state. This is just good test design and should be possible for nearly all cases (I think I've had to do this once so far for all the tests I've written)
1. If you do need to - consider instead refactoring the component such that you don't need to modify internal state to achieve your testing goals
1. But in the end we have to balance time spent writing tests with time spent refactoring and the amount we gain from that time spent. And if it ends up being a lot of work for very little gain then I personally see no problem with having a few places expose their internals (or have tests access/modify them).

In this specific case I see way too many private -> protected changes for my liking. I would very much question why we aren't able to test this behavior without making these protected. So @ranasaria if you really believe that these are necessary please go into detail explaining why so people have context and can perhaps suggest alternatives.

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts (hunk 8 above)
Comment (aaomidi) [reply src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:387]: The only time you'd want to do this is if you want to handle potential exceptions here instead of at the caller.

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts (hunk 13 above)
Comment (ranasaria) [reply src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:440]: So these are the two questions that we need to address here: 
1. Should we ever need access to private members in tests? 
2. And if we fathom that it is indeed ok to allow access to private members in testing, then what is the best way to do so in typescript.

Let's address the 1st question. Let me present an example of a common code pattern from our codebase:
1. There exists an EventEmitter on class Foo and its event is available publicly for listeners to subscribe to.
2. The class under test, say 'Bar', subscribes to that public event on Foo and registers an async private method to be invoked when the corresponding emitter fires. The code in Bar or the product does not care when the execution of this async method completes and it raises errors when it runs into any unexpected issues.

So to get this code tested, we can write a test to invoke whatever code will fire the event in question by using the corresponding EventEmitter in Foo class. However, to do verification of the effects of the async private method that is the focus of this test, we need to be able to wait for the completion of this async method. There is no current way available to do this wait. The actual product code does not need to do this wait because there is nothing that needs to be done in the product after that method finishes. By doing access to the private method, one can test this code out in two parts:
1. Write test code to make sure that the private method under test gets invoked when the EventEmitter on Foo fires. 
2. Invoke the async method through direct access so you can await on it and do verification once the method execution has completed.

Let's also address the 2nd question: (this may be redundant if we find a good way to solve the first question without requiring direct access). So if were to allow direct access to private members then what is the best way to do so. Two obvious ways come to mind:
1. Direct member access using string indexers on the corresponding object. 
2. Exposing private members as protected to provide access via a child class.

I have found the 2nd approach to definitely be better because it makes a future developer enhancing the class under test ('Bar, in our above example) be aware of the protected members and potential usages outside of that class and hence allows him/her to do the right thing. Of course, the downside is that the developer still has to worry about the protected members which s/he would not have had to if it had remained private. The first approach actually hides usage of those members completely making life hard for the future developer who is likely to discover breakages of tests later than sooner.

File: src/sql/workbench/contrib/notebook/test/testCommon.ts (hunk 12 above)
Comment (ranasaria) [reply src/sql/workbench/contrib/notebook/test/testCommon.ts:26]: it is used in several places. I find this a useful shortcut as I find myself doing this often. Would it be better to put this in my own test class?

Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/notebook/test/testCommon.ts:26]: Well unless I'm mistaken you're not using this in this PR at all. So it should be removed unless you actually use it. 

And I don't see what benefits this brings vs just creating the mocks inline. Most of these options we leave as the default anyways and you can access the object itself directly from the mock so this isn't really saving much that I can see. 

(I also would suggest you don't use spiedObject as the name - spies are a separate thing than mocks)

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts (hunk 13 above)
Comment (Charles-Gagnon) [reply src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts:440]: @ranasaria Could you please point to an example in the code/tests you wrote that shows where this was needed to be done? It'd help to give direct context to everyone so we don't have to spend time going through your tests to understand what parts you're referring to.

File: src/sql/workbench/contrib/notebook/test/testCommon.ts (hunk 12 above)
Comment (ranasaria) [reply src/sql/workbench/contrib/notebook/test/testCommon.ts:26]: Removed this method

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts (hunk 8 above)
Comment (ranasaria) [reply src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts]: reverted

File: src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts (hunk 13 above)
Comment (ranasaria) [reply src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts]: Fixed. Removed the Subclass altogether.

File: src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts [hunk 15]
```diff
@@ -4,52 +4,152 @@
...
-		didInstallEvent: Emitter<DidInstallExtensionEvent>,
-		uninstallEvent: Emitter<IExtensionIdentifier>,
-		didUninstallEvent: Emitter<DidUninstallExtensionEvent>;
+suite('Test class ProviderDescriptor:', () => {
+	test('Verifies varies getter setters of Provider Descriptor', async () => {
+		const notebookProvider = <INotebookProvider>{};
+		const providerDescriptor = new ProviderDescriptor(notebookProvider);
+		assert.strictEqual(providerDescriptor.instance, notebookProvider, `providerDescriptor instance should be the value passed into the constructor`);
+		const providerInstancePromise = providerDescriptor.instanceReady;
+		assert.notStrictEqual(providerInstancePromise, undefined, `providerDescriptor instanceReady should return an undefined promise object`);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts:85]: You say it shouldn't return an undefined object but you're checking that it's NOT an undefined object

File: src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts [hunk 16]
```diff
@@ -4,52 +4,152 @@
...
+	handleNotebookClosed(_notebookUri: URI): void {
+		// do nothing
+	}
+}
 
-	let installEvent: Emitter<InstallExtensionEvent>,
-		didInstallEvent: Emitter<DidInstallExtensionEvent>,
-		uninstallEvent: Emitter<IExtensionIdentifier>,
-		didUninstallEvent: Emitter<DidUninstallExtensionEvent>;
+suite('Test class ProviderDescriptor:', () => {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts:79]: You don't need to prefix every suite with `Test class` - we know this is a test already. Just the name of the thing being tested is fine for the top level suite.

File: src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts [hunk 17]
```diff
@@ -85,4 +186,390 @@ suite('Notebook Service Tests', function (): void {
...
+		const providerId = 'providerId1';
+		createRegisteredProviderWithManager({ notebookService, providerId });
+		notebookService.registerProvider(providerId, undefined);
+		//verify method under test logs error and does not throw when extensionService.whenInstalledExtensionRegistered() throws
+		const error: Error = new Error('Extension Registration Failed');
+		extensionServiceMock.setup(x => x.whenInstalledExtensionsRegistered()).throws(error);
+
+		logServiceMock.setup(x => x.error(TypeMoq.It.isAny()))
+			.returns((_error: string | Error, ...args: any[]) => {
+				assert.ok(_error instanceof Error, `error passed into logService.error call should be an instance of ${Error}`);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts:206]: Just the strictEquals check here is enough - we already know that's an Error

File: src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts [hunk 18]
```diff
@@ -85,4 +186,390 @@ suite('Notebook Service Tests', function (): void {
...
+				assert.strictEqual(_error, error, `error object passed to logService.error call must be the one thrown from whenInstalledExtensionsRegistered call`);
+			})
+			.verifiable(TypeMoq.Times.once());
+		await notebookService.getOrCreateNotebookManager(providerId, URI.parse('untitled:uri1'));
+		logServiceMock.verifyAll();
+	});
+
+
+	test('verify that getProviderInstance throws when no providers are registered', async () => {
+		const methodName = 'getProviderInstance';
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts:216]: Just get rid of these kind of vars IMO - they don't really provide any use vs just embedding in the strings directly...

File: src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts [hunk 19]
```diff
@@ -85,4 +186,390 @@ suite('Notebook Service Tests', function (): void {
...
+				assert.ok(_error instanceof Error, `error passed into logService.error call should be an instance of ${Error}`);
+				assert.strictEqual(_error, error, `error object passed to logService.error call must be the one thrown from whenInstalledExtensionsRegistered call`);
+			})
+			.verifiable(TypeMoq.Times.once());
+		await notebookService.getOrCreateNotebookManager(providerId, URI.parse('untitled:uri1'));
+		logServiceMock.verifyAll();
+	});
+
+
+	test('verify that getProviderInstance throws when no providers are registered', async () => {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts:215]: I'd update the name - you're not testing getProviderInstance directly here so the title isn't accurate

File: src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts [hunk 20]
```diff
@@ -85,4 +186,390 @@ suite('Notebook Service Tests', function (): void {
...
+		const result1Again = await notebookService.getOrCreateNotebookManager(provider1Id, uri1);
+		assert.strictEqual(result2, provider2Manager, `the notebook manager for the provider must be the one returned by getNotebookManager of the provider`);
+		assert.notStrictEqual(result1, result2, `different notebookManagers should be returned for different uris`);
+		assert.strictEqual(result1, result1Again, `same notebookManagers should be returned for same uri for builtin providers`);
+		const result2Again = await notebookService.getOrCreateNotebookManager(provider2Id, uri2);
+		assert.strictEqual(result2, result2Again, `same notebookManagers should be returned for same uri for custom providers`);
+	});
+
+	test('verifies add/remove/find/list/renameNotebookEditor methods and corresponding events', async () => {
+		assert.strictEqual(notebookService.findNotebookEditor(undefined), undefined, `findNotebookEditor should return the editor when searching with the latest Uri`);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts:282]: Message seems wrong - you're checking that it's undefined

File: src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts (hunk 15 above)
Comment (ranasaria) [reply src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts:85]: will fix the comment

File: src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts (hunk 17 above)
Comment (ranasaria) [reply src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts:206]: the error from the call signature could have been a string so this ensures that an instance of Error was thrown right?

File: src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts (hunk 20 above)
Comment (ranasaria) [reply src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts:282]: yes will fix message - good catch.

File: src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts (hunk 18 above)
Comment (ranasaria) [reply src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts:216]: if it is used once then I agree, but in most cases, I was referring to the methodName in the verification message so wanted a const for it instead of repeating the same string.

File: src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts (hunk 16 above)
Comment (ranasaria) [reply src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts:79]: Fixed

File: src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts (hunk 15 above)
Comment (ranasaria) [reply src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts:85]: fixed

File: src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts (hunk 20 above)
Comment (ranasaria) [reply src/sql/workbench/contrib/notebook/test/browser/notebookService.test.ts:282]: fixed.

**Comment by ranasaria** [comment]
There is a test failure in extensin test execution. This failure looks unrelated to my changes. The tests passed in a previous run this afternoon. There have been no product code changes since that run. I have run extension tests locally on this branch and they have passed. So I am merging in this code with my finger crossed.

**Comment by Charles-Gagnon** [comment]
@ranasaria In that case you should have re-ran the checks.

In this case the failure is a known issue that Aasim is fixing so it should be fine, but in general it would be good not to get in the habit of merging in tests when you have test failures in the checks.
